src/main/java/uk/gov/hmcts/sptribs/common/event/page/ApplicantDetails.java

```diff
@@ -28,4 +34,25 @@ public void addTo(PageBuilder pageBuilder) {
             .mandatory(CicCase::getApplicantEmailAddress, "cicCaseApplicantContactDetailsPreference = \"Email\"")
             .done();
     }
+
+    private AboutToStartOrSubmitResponse<CaseData, State> midEvent(
+        CaseDetails<CaseData, State> details,
+        CaseDetails<CaseData, State> detailsBefore) {
+        final CaseData data = details.getData();
+        final List<String> errors = new ArrayList<>();
+
+        if (null != data.getCicCase() && null != data.getCicCase().getApplicantAddress()) {
+            if (StringUtils.isEmpty(data.getCicCase().getApplicantAddress().getCountry())) {
```

Comment:
If ApplicantAddress is null, does that mean there should be an error added? Just a thought.

Reply:
No, it can address or email. We need to check for Country/post code only if Address is selected.